lib/torrent.js

```diff
@@ -403,7 +408,55 @@ Torrent.prototype._onMetadata = function (metadata) {
   })
 
   self._debug('verifying existing torrent data')
-  parallelLimit(self.pieces.map(function (piece, index) {
+  if (self._fileModtimes && self._store === FSChunkStore) {
+    // don't verify if the files haven't been modified since we last checked
+    self.getFileModtimes(function (err, fileModtimes) {
+      if (err) return self._onError(err)
+
+      var unchanged = self.files.map(function (_, index) {
+        return fileModtimes[index] === self._fileModtimes[index]
+      }).reduce(function (a, b) {
+        return a && b
+      })
```

Comment:
This will probably crash when there's only one file, since a default starting value was not specified for reduce.